spec/models/product_spec.rb

```diff
@@ -0,0 +1,181 @@
+require "rails_helper"
+
+RSpec.describe Product, type: :model do
+  describe "Associations" do
+    it{should belong_to(:category)}
+    it{should have_many(:product_details).dependent(:destroy)}
+    it{should have_many(:comments).dependent(:destroy)}
+    it{should have_many(:rates).dependent(:destroy)}
+    it{should have_many(:likes).dependent(:destroy)}
+  end
+
+  describe "Scopes" do
+    let!(:caterories){FactoryBot.create_list :category, 2}
+    let!(:product_sizes){FactoryBot.create_list :product_size, 2}
+    let!(:product_colors){FactoryBot.create_list :product_color, 2}
+    let!(:users){FactoryBot.create_list :user, 3}
+
+    let!(:product1){FactoryBot.create :product, name: "iphone"}
+    let!(:product2){FactoryBot.create :product, name: "samsung"}
+    let!(:product3){FactoryBot.create :product, name: "lg"}
+
+    let!(:product_details1) do
+      FactoryBot.create_list :product_detail, 4, product_id: product1.id,
+                              product_size_id: product_sizes[0].id,
+                              product_color_id: product_colors[0].id,
+                              cost: 1_000_000
+    end
+
+    let!(:product_details2) do
+      FactoryBot.create_list :product_detail, 4, product_id: product2.id,
+                              product_size_id: product_sizes[1].id,
+                              product_color_id: product_colors[1].id,
+                              cost: 2_000_000
+    end
+
+    let!(:product_details3) do
+      FactoryBot.create_list :product_detail, 4, product_id: product3.id,
+                              product_size_id: product_sizes[0].id,
+                              product_color_id: product_colors[1].id,
+                              cost: 3_000_000
+    end
+
+    let!(:order1){FactoryBot.create :order, user_id: users[0].id}
+    let!(:order2){FactoryBot.create :order, user_id: users[1].id}
+    let!(:order3){FactoryBot.create :order, user_id: users[1].id}
+
+    let!(:order_details1) do
+      FactoryBot.create_list :order_detail, 4,
+                             quantity: 10,
+                             product_detail_id: product_details1[0].id,
+                             order_id: order1.id
+    end
+
+    let!(:order_details2) do
+      FactoryBot.create_list :order_detail, 4,
+                             quantity: 20,
+                             product_detail_id: product_details2[0].id,
+                             order_id: order2.id
+    end
+
+    let!(:order_details3) do
+      FactoryBot.create_list :order_detail, 4,
+                              quantity: 30,
+                              product_detail_id: product_details3[0].id,
+                              order_id: order3.id
+    end
```

Comment:
mớ code này chỉ dùng cho `describe ".order_created_at" do` thì cần j khai báo trên ni em

Reply:
Dạ vâng ạ, không cần ` describe "sort by products creation date" do` và `it "has 3 products sorted by creation date" do` ạ